scripts/pre_commit_linter.py

```diff
@@ -342,10 +356,48 @@ def _pre_commit_linter():
     summary_messages.append(py_result.get())
     print '\n'.join(summary_messages)
     print ''
+    return all_files, summary_messages
 
+
+# This part is used for detecting bad patterns.
+def _check_bad_patterns(all_files, summary_messages):
+    print 'Starting Pattern Checks'
+    print '----------------------------------------'
+    total_files_checked = 0
+    total_error_count = 0
+    all_files = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    for filename in all_files:
+        with open(filename) as f:
+            content = f.read()
+            total_files_checked += 1
+            for pattern in BAD_PATTERNS:
+                if pattern in content:
+                    summary_messages.append(_MESSAGE_TYPE_FAILED)
+                    print '%s --> %s' % (
+                        filename, BAD_PATTERNS[pattern])
+                    total_error_count += 1
+
+    print ''
+    print '----------------------------------------'
+    print ''
+    if total_files_checked == 0:
+        print "There are no files to be checked."
+    else:
+        print '(%s files checked, %s errors found)' % (
+            total_files_checked, total_error_count)
+
+    return summary_messages
+
+
+def main():
+    all_files, summary_messages = _pre_commit_linter()
```

Comment:
With regards to technical design, the separation of responsibilities here seems incorrect. The following makes more conceptual sense, I think:

There should be a method which gets a list of all files to run checks on.
Then, _pre_commit_linter() should be run on this list.
Then, _check_bad_patterns() should be run on this list.

Can you refactor accordingly?